tests/ui/asm/x86_64/type-check-4.rs

```diff
@@ -1,14 +1,12 @@
 //@ only-x86_64
 //@ compile-flags: -C target-feature=+avx512f
-
-use std::arch::{asm, global_asm};
+//@ check-pass
 
 use std::arch::x86_64::{_mm256_setzero_ps, _mm_setzero_ps};
+use std::arch::{asm, global_asm};
 
 fn main() {}
 
-// Constants must be... constant
-
 static S: i32 = 1;
```

Comment:
To preserve the original test as closely as possible, I'd suggest to *not* make this check-pass. Instead, make this here `static mut` so that there continues to be a const-eval error in this test. (Same for the aarch test.)

Reply:
Applied

Reply:
These will also need an `unsafe` block.

Reply:
Huh it looks like `global_asm!` macro accepted the constant wrapped in `unsafe`. Should we reject it?

Specifically this passed the `check` 🤔 .
```rust
global_asm!("{}", const unsafe { S });
```

I feel like we should rejected. I will prepare a patch to reject it if that should be the case.

Reply:
What is `S` here?

If it is a `static mut`, there seems to be a deeper issue where the usual const rules are not applied. That should not be papered over with a new check that specifically covers this case, we need to instead make sure the existing checks are applied here as well.

Reply:
If this is with a `static mut`, please file an issue with a self-contained example demonstrating the problem. That should have its own discussion, to be resolved before we can proceed with stabilization.

Reply:
That was a close one.

For some reason, tests in `tests/ui/asm` are run with `--emit=metadata`, so that `global_asm!` was not checked at all for some reason. I added a flag `--emit=asm` and have gotten back the errors on use of `static mut`.

Reply:
Switched to `//@ build-fail`.